### DISCUSSION TITLE
Test error - unable to place monster due to player in the way

### DISCUSSION THREAD

**Discussion by actual-nh** [CleverRaven/Cataclysm-DDA#46476] [General]
Hi. #46473 has gotten some potentially-useful results (see #46439 for others) from [the first Github g++ build](https://github.com/CleverRaven/Cataclysm-DDA/pull/46473/checks?check_run_id=1635456183). Could someone with a Linux box access this and take a closer look?

**Comment by anothersimulacrum** [comment]
Travis has a crash, github seems fine.
The github build doesn't work for me (different ncurses libs), but I can run with the same parameters as the travis build and reproduce the crash.
`--min-duration 0.2 --use-colour yes --rng-seed 1609558037  --order rand`

This bypasses it:
```diff
diff --git a/tests/vehicle_ramp_test.cpp b/tests/vehicle_ramp_test.cpp
index 7a587923f9..81a561951d 100644
--- a/tests/vehicle_ramp_test.cpp
+++ b/tests/vehicle_ramp_test.cpp
@@ -273,6 +273,7 @@ static void level_out( const vproto_id &veh_id, const bool drop_pos )
     REQUIRE( z_span.size() > 1 );

     monster *dmon_p = g->place_critter_at( mtype_id( "debug_mon" ), map_starting_point );
+    REQUIRE( dmon_p );
     monster &dmon = *dmon_p;

     for( int y = 0; y < SEEY * MAPSIZE; y++ ) {
 ```

**Reply by actual-nh** [reply]
Thanks! I will try that.

**Reply by anothersimulacrum** [reply]
It doesn't fix it, it just turns the crash into a failure.

**Comment by actual-nh** [comment]
That's OK, at least now we'll have a better idea of what's going wrong, and will also be able to check on other tests a bit more. I've (just) found a combination of tests that - unlike running just the vehicle_level_test - result in the error without having to run ~half the tests.

**Comment by kevingranade** [comment]
Looks like something is occupying the tile where the test is trying to place the monster? Should just require better map clearing.

**Reply by anothersimulacrum** [reply]
It's the player, they need to be moved.
